src/lib/when.ts

```diff
@@ -0,0 +1,86 @@
+import * as LRU from 'lru-cache';
+import {Updatable} from './sparse-map';
+
+const proxyCache = LRU(64);
+
+function isObject(o: any): Boolean {
+  return o === Object(o);
+}
+
+const EMPTY_FN = () => {};
+export class SelfDescribingProxyHandler {
+  name: string;
+
+  constructor(name: string) {
```

Comment:
you can simplify 

```js
class A {
 public a: string;
 construnctor(a: string) {
  this.a = a;
 }
}
```

into 
```js
class A {
 constructor(public a: string) //can do it for private, public readonly, etcs..
}
```

just in case if you'd like shorthand representation.